hera_cal/smooth_cal.py

```diff
@@ -682,21 +824,72 @@ def time_freq_2D_filter(self, freq_scale=10.0, time_scale=1800.0, tol=1e-09,
                         and fringe rate = 0
                 'plus': produce a separable calibration solution by only keeping modes with 0 delay,
                         0 fringe rate, or both
-            window: window function for filtering applied to the frequency axis.
+            window: window function for filtering applied to the frequency axis. Only used when the method is 'CLEAN'
                 See aipy.dsp.gen_window for options.
             maxiter: Maximum number of iterations for aipy.deconv.clean to converge.
+            eigenval_cutoff: sinc_matrix eigenvalue cutoffs to use for included dpss modes.
+                Only used when the filtering method is 'DPSS'
+            method: Algorithm used to smooth calibration solutions. Either 'CLEAN' or 'DPSS':
+                'CLEAN': uses the CLEAN algorithm to
+                'DPSS': uses discrete prolate spheroidal sequences to filter calibration solutions
+            skip_flagged_edges : if True, do not filter over flagged edge times (filter over sub-region)
+                default is False, only used when method='DPSS'
             win_kwargs : any keyword arguments for the window function selection in aipy.dsp.gen_window.
                 Currently, the only window that takes a kwarg is the tukey window with a alpha=0.5 default
         '''
+        if method == 'DPSS':
```

Comment:
Could you avoid doing all this by just doing what you do for sol_matrix (`sol_matrix = info['sol_matrix'] if np.allclose(wgts_grid, wgts_prev) else None`) for the design matrix too?

Reply:
I decided to simplify the function a little. If you're using `skip_flagged_edges`, your weights/gain grid may come out to be different sizes depending on the flagging pattern, which would mean you would need to recalculate the DPSS vectors to fit the new size of the gain grid. Calculating the DPSS filters can start to take a while as you get larger gain grid sizes, so I was using this as a way of keeping track of the grid sizes we've already calculated the DPSS vectors for. There might be a way to better optimize this in the future, but for now, I've just removed it for simplicity and readability.